### ISSUE TITLE
MA Thesis on Word Sense Embeddings - Description of the Initial Experiment

### ISSUE THREAD

**Issue by alexanderpanchenko** [tudarmstadt-lt/sensegram#2]

**Comment by mpelevina** [comment]
Initial experiment on the 1.5 wiki model is ready
Results are posted to this table: https://docs.google.com/spreadsheets/d/1VaniexahrFJmKyd1XmqeTSjgekerV67CyJ0biP_TAdk/edit?ts=56700d33
All models and output files are on frink: /home/pelevina/experiment
Later I'll reorganise the files in this folder and create a good description folder system and files meaning

**Comment by alexanderpanchenko** [comment]
great. pls include extra parameters of the model in the table (corpus lemmatized or not, parameters of CWhispers, pooling method, disambiguation method).

the results look much better than another model we have. this is good, but before reporting it "upstairs" we need to make sure that there is no flow in the computations. so pls double check the results. 

otherwise, train now on the full wikipedia. please consider this simple trick to make computation of NNs faster:

0. get vocabulary of your word2vec model and save in a file.
1. use the ```split``` unix command to split it into 12 parts. 
2. run 12 instances of your program each on different part of the vocabulary thus using 12 cores.
3. merge the results with the ```cat``` unix command

**Comment by alexanderpanchenko** [comment]
in addition, please evaluate on SemEval 2013. use this script: 

https://github.com/tudarmstadt-lt/contextualization-eval/blob/master/semeval_2013_13/evaluate.sh

**Comment by mpelevina** [comment]
Great!
And thank you for the tip. 
About double checking the results, could you suggest some examples of what the flows could be and how to look for them? Typical bugs or anything else? 
SemEval2013 has a different test set which I'll have to run over with this model before evaluating the results? It has nothing to do with data and formats used for TWSI evaluation, am I right?

**Comment by alexanderpanchenko** [comment]
yes, the first thing to check is if the model actually non-trivial. if it assigns always the same sense this is bad.

**Comment by alexanderpanchenko** [comment]
otherwise, a good debugging would be to fill the related words column in the training set and compare it visually with the golden related words.

**Comment by alexanderpanchenko** [comment]
no, semeval 2013 format is exactly the same

Dataset-SemEval-2013-13.csv

sorry, you need to use this script instead: https://github.com/tudarmstadt-lt/contextualization-eval/blob/master/semeval_2013_13.sh

**Comment by alexanderpanchenko** [comment]
> SenseGram (not lemmatized; CW param: -n200, -N200, cw, -e 0.01; pooling: mean of cluster vectors;)

please also add information about minimum cluster size and disambiguation method.

**Comment by alexanderpanchenko** [comment]
> -output ../experiment/model/12wiki_vectors.bin -cbow 1 -size 200 -window 8 -negative 25 -hs 0 -sample 1e-4 -threads 20 -binary 1 -iter 15

do you use CBOW or skip-grams?

**Comment by alexanderpanchenko** [comment]
> Results are posted to this table: https://docs.google.com/

pls do not post Google docs links here. the table is publicly available

**Comment by mpelevina** [comment]
I'm using CBOW. 
I haven't given it much thought at that moment and it was a default parameter. Should we train with skip-grams?

**Comment by alexanderpanchenko** [comment]
usually skip-gram model with context window size around 5 gives better results. check: dialog-21.ru/digests/dialog2015/materials/pdf/ArefyevNVetal.pdf

**Comment by Sandy4321** [comment]
Cool, where is it to read
